plugin/aws/ec2Dir.go

```diff
@@ -17,12 +17,18 @@ type ec2Dir struct {
 	entries []plugin.Entry
 }
 
-func newEC2Dir(session *session.Session) *ec2Dir {
+func ec2DirTemplate() *ec2Dir {
```

Comment:
These `Template` functions are here because Go doesn't support (simple) meta-programming. In Ruby, this would look something like
```
class EC2Dir < Wash::Entry
  singleton "ec2"
  parent_of EC2InstancesDir

  def list
  end

  ...
end
```

where `Wash::Entry` would have a `schema` method that would check for these class-level tags and construct the schema accordingly (that's also how external plugins would support `schema` -- the entries would include an implemented "schema" method. We'll need to handle the case when entries don't implement `schema` to make this easy for Bash, or maybe Bash plugins can't access the schema functionality).

If/when we add attributes to this list, we'd likely have `ec2Dir.SupportedAttributes("ctime", "mtime")` which would translate to class-level `ctime`, `mtime` tags in Ruby, so e.g. something like
```
class EC2Dir < Wash::Entry
  attributes :ctime, :mtime
end
```
(which would then be `@ctime` and `@mtime` fields in the class).

Reply:
Think we could include the cache TTLs in the schema, since they're also class-level params. Would be useful if someone's curious about how long things are cached in Wash.

Reply:
We could also include meta schemas here to validate meta primary queries. Idea is to have a method that takes in an empty object/struct, marshal it (for the purpose of obtaining its fields), and work from there. Doing this could also remove the (annoying) behavior of defaulting maxdepth to 1, and having to set it to -1 for meta to recurse.